lib/IntegrationEvent/Implementation/AWS-SNS-SQS/SQSHelper.ts

```diff
@@ -60,24 +68,7 @@ export class SQSHelper {
           const subscriptions = options.getSubscriptions();
 
           // Put discover logic in here to support dynamically add more subscriptions through applications.
-          const subscription = [...subscriptions.entries()].find(
-            ([eventConstructor]) => eventConstructor.name === parsedBody.EventName
-          );
-
-          if (!subscription) {
-            throw new Error(`There is no subscription respecting to this event: ${parsedBody.EventName}.`);
-          }
-
-          // Construct event entity
-          const [eventConstructor, eventHandlers] = subscription;
-          if (!eventConstructor || !eventHandlers || eventHandlers.length === 0) {
-            throw new Error(`There is no event handler respecting to this event: ${parsedBody.EventName}.`);
-          } else {
-            const event = new eventConstructor(parsedBody.EventBody);
-            for (const handler of eventHandlers) {
-              await handler.handle(event);
-            }
-          }
+          await SQSHelper.handleEvent(parsedBody, subscriptions, options.RedisClient);
```

Comment:
Maybe add a try-catch around this? Otherwise if something goes wrong with event handling it would be an unhandled exception with little debugging details right?

Reply:
We will only be able to console.log the error here, unless we add logger or slack integration. We have try-catch in all of our event handler classes in the backend. Wouldn't the error handling be more scenario-specific there?

Reply:
try-catch in event handler classes is more of a good practice, it is possible we overlook it and forgot to try catch. Having it here acts as a global try-catch for all event handlers in case something errors got thrown out. console.log()errors will go to PM2 logs, which still helps us with debugging.

Reply:
Added in latest commit